upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -782,6 +792,16 @@ func setupNodes(opt *NewClusterOptions, cluster *api.Cluster, zoneToSubnetMap ma
 			g.Spec.Zones = []string{zone}
 		}
 
+		if api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderAWS && (g.Spec.InstanceMetadata == nil || g.Spec.InstanceMetadata.HTTPTokens == nil) {
+			k8sVersion, err := version.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+			if err == nil && version.IsKubernetesGTE("1.20", *k8sVersion) {
+				g.Spec.InstanceMetadata = &api.InstanceMetadataOptions{
```

Comment:
this will overwrite InstanceMetadata if it already exists but HTTPTokens is nil. We should probably only create InstanceMetadata if it is nil, otherwise just set HTTPTokens and HTTPPutResponseHopLimit if they arent set.

Reply:
Thanks. I think this last change should do it.